packages/ai/src/ai/common/sandbox.py

```
@@ -299,16 +300,28 @@ def _run() -> None:
-        timeout=60,
-    )
-    # Clear the import cache so the freshly installed module is found
-    importlib.invalidate_caches()
+    if not isinstance(package, str) or not package:
+        raise ValueError(f'Invalid package name: {package!r}')
+    if len(package) > _MAX_PACKAGE_NAME_LEN:
+        raise ValueError(f'Package name too long ({len(package)} chars, max {_MAX_PACKAGE_NAME_LEN})')
+    if not _VALID_PACKAGE_RE.match(package):
+        raise ValueError(f'Invalid package name {package!r}: must contain only alphanumeric characters, dots, hyphens, and underscores, and must start/end with an alphanumeric character')
+    with _pip_lock:
+        subprocess.check_call(
```

Ruff (0.15.7)

[error] `subprocess` call: check for execution of untrusted input

(S603)